codebase2/util/U/Util/Cache.hs

```diff
@@ -4,19 +4,26 @@
 module U.Util.Cache where
 
 import Prelude hiding (lookup)
+import Control.Monad.IO.Class (liftIO)
 import UnliftIO (MonadIO, newTVarIO, modifyTVar', writeTVar, atomically, readTVar, readTVarIO)
 import qualified Data.Map as Map
 import Data.Functor (($>))
 import Control.Monad (when)
 import Data.Foldable (for_)
 
-data Cache m k v =
-  Cache { lookup :: k -> m (Maybe v)
-        , insert :: k -> v -> m ()
+data Cache k v =
+  Cache { lookup_ :: k -> IO (Maybe v)
+        , insert_ :: k -> v -> IO ()
         }
```

Comment:
This is the main change - `Cache k v` rather than `Cache m k v`. With this scheme, the cache just needs to be _used_ in contexts that permit `IO`. 

It's totally fine for the cache to be used in multiple contexts, each with different `MonadIO m`.

This is much nicer than having to pick a single `m` for the `Cache` and then needing to lift or whatever as appropriate depending on the context(s) where it's being used.